hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/group/commands/CreateGroupRepositoryCommand.java

```diff
@@ -0,0 +1,37 @@
+package pl.allegro.tech.hermes.management.domain.group.commands;
+
+import pl.allegro.tech.hermes.api.Group;
+import pl.allegro.tech.hermes.domain.group.GroupRepository;
+import pl.allegro.tech.hermes.management.domain.dc.RepositoryCommand;
+
+public class CreateGroupRepositoryCommand extends RepositoryCommand<GroupRepository> {
+
+    private final Group group;
+
+    public CreateGroupRepositoryCommand(Group group) {
+        this.group = group;
+    }
+
+    @Override
+    public void backup(GroupRepository repository) {}
+
+    @Override
+    public void execute(GroupRepository repository) {
+        repository.createGroup(group);
+    }
+
+    @Override
+    public void rollback(GroupRepository repository) {
+        repository.removeGroup(group.getGroupName());
+    }
+
+    @Override
+    public Class<GroupRepository> getRepositoryType() {
+        return GroupRepository.class;
+    }
+
+    @Override
+    public String toString() {
+        return "CreateGroup(" + group.getGroupName() + ")";
```

Comment:
I guess, those `toString()` methods for commands could be done automatically using class name (substring etc.), placed in parent `RepositoryCommand` maybe. :)

Something like:

```java
// base class
private final static int NAME_LENGTH = RepositoryCommand.class.getSimpleName().length();

protected String printableName(String id) {
    String name = this.getClass().getSimpleName();
    return name.substring(0, name.length() - NAME_LENGTH) + "(" + id + ")";
}

@Override
public String toString() {
    return printableName(""); // default without an identifier, not sure if needed at all
}
```

```java
// command class
@Override
public String toString() {
    return printableName(provider.getName());
}
```

Reply:
Good point, but I would like to leave these 16 classes as they're :P

Reply:
cool :)